include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -223,6 +223,8 @@ class argument_parser
             throw parser_design_error("Option Identifier '" + long_id + "' was already used before.");
         if (short_id == '\0' && long_id.empty())
             throw parser_design_error("Option Identifiers cannot both be empty.");
+        if (id_forbidden(std::string(1, short_id)) || id_forbidden(long_id))
```

Comment:
This is a nice idea to check this but I think it might be easier to just insert the forbidden ids into the `used_option_ids` set? It would avoid having this extra if clause and the ones below. 
Add this in line 570 of `argument_parser.hpp` and see if it works. 
```cpp
used_option_ids.insert({"h", "help", "hh", ... });
```

Reply:
Do you think it might be confusing if the error message is the same for the "forbidden IDs" and for the user set IDs?

Reply:
Also, could we just instantiate used_option_ids with "h", "help", etc.? As such:
```cpp
std::set<std::string> used_option_ids{"h", "help", ...};
```

Reply:
> Do you think it might be confusing if the error message is the same for the "forbidden IDs" and for the user set IDs?

I think it's fine because it will tell the user that `an option with identifier -h is already taken`. Which is maybe even more true than just saying its forbidden since there IS an option -h for their program? What do you think?

 > Also, could we just instantiate used_option_ids with "h", "help", etc.? As such:

Yes!